content/blog/9-30-24-go-wsl-nvim.md

```diff
@@ -0,0 +1,241 @@
+## Introduction
+
+As a minimalist at heart, I've always been drawn to simplicity - that's part of what attracted me to Go. Years of skateboarding and wrist injuries, combined with my ADHD-induced tendency to get distracted by complex IDEs, led me to explore Vim motions and Neovim. While my ultimate goal is to run ArchLinux, blast some breakcore, and hack away on Neovim like a true power user, I'm not quite there yet.As a Windows user, I wanted to find a middle ground - a way to use Neovim and Go without completely abandoning my current setup. Enter WSL (Windows Subsystem for Linux). This guide is my first attempt at setting up a development environment that combines the best of both worlds: Neovim and Go on WSL, with the added ability to debug and step through code.Fair warning: this is just the beginning. Expect more blogs as I continue to refine and expand my Neovim setup. For now, let's dive into setting up WSL, Go, and the Debug Adapter Protocol (DAP) with Kickstart.nvim. We'll cover everything from WSL installation to configuring Neovim for Go development and debugging.
+## Step 1: Installing WSL
+
+1. Open PowerShell as Administrator and run:
+   ```
+   wsl --install
+   ```
+2. Restart your computer when prompted.
+3. After restart, open Ubuntu from the Start menu to complete the setup.
+
+## Step 2: Installing Go
+
+1. Update your package list:
+   ```
+   sudo apt update
+   ```
+2. Remove any existing Go installation:
+   ```
+   sudo rm -rf /usr/local/go
+   ```
+3. Download Go 1.22.7, you can use whatever go version you want here
+   ```
+   wget https://go.dev/dl/go1.22.7.linux-amd64.tar.gz
+   ```
+4. Extract Go to /usr/local:
+   ```
+   sudo tar -C /usr/local -xzf go1.22.7.linux-amd64.tar.gz
+   ```
+5. Set up your Go environment:
+   ```
+   echo 'export GOROOT=/usr/local/go' >> ~/.bashrc
+   echo 'export PATH=$GOROOT/bin:$PATH' >> ~/.bashrc
+   source ~/.bashrc
+   ```
+6. Verify the installation:
+   ```
+   go version
+   ```
+
+## Step 3: Installing Neovim
+
+1. Download the latest Neovim AppImage:
+   ```
+   curl -LO https://github.com/neovim/neovim/releases/latest/download/nvim.appimage
+   ```
+2. Make it executable:
+   ```
+   chmod u+x nvim.appimage
+   ```
+3. Move it to a directory in your PATH:
+   ```
+   sudo mv nvim.appimage /usr/local/bin/nvim
+   ```
+
+## Step 4: Setting Up Kickstart.nvim
+
+1. Before we get started here be sure to checkout my (dotfiles)[https://github.com/williycole/dotfiles] incase you want to see the final result or need a reference along the way. Also feel free to fork/clone them. 
```

Comment:
nit: adding comma after 'Before we get started here',
spacing: incase -> in case